src/main/java/seedu/address/logic/parser/SearchCommandParser.java

```diff
@@ -86,23 +97,54 @@ public SearchCommand parse(String args) throws ParseException {
      * @param value The value associated with the {@code prefix} to check.
      * @throws ParseException If the prefix is present but its value is empty.
      */
-    private void checkForEmptyInput(ArgumentMultimap argMultimap, Prefix prefix, String value)
+    private void validatePrefixNotEmpty(ArgumentMultimap argMultimap, Prefix prefix, String value)
             throws ParseException {
         if (argMultimap.getValue(prefix).isPresent() && value.trim().isEmpty()) {
             throw new ParseException(MESSAGE_EMPTY_SEARCH_PREFIX);
         }
     }
 
-    /**
-     * Combines a new predicate with the existing combined predicate using a logical AND.
-     * If the combined predicate is null, it initializes it with the new predicate.
-     *
-     * @param combinedPredicate The existing combined predicate.
-     * @param newPredicate The new predicate to add.
-     * @return The updated combined predicate.
-     */
-    private Predicate<Person> combinePredicate(Predicate<Person> combinedPredicate, Predicate<Person> newPredicate) {
-        return combinedPredicate == null ? newPredicate : combinedPredicate.and(newPredicate);
+    private Predicate<Person> createNamePredicate(String nameArgs) {
+        if (nameArgs.isEmpty()) {
+            return null;
+        }
+        return new NameContainsKeywordsPredicate(Arrays.asList(nameArgs.split("\\s+")));
+    }
+
+    private Predicate<Person> createTagPredicate(String tagArgs) {
+        if (tagArgs.isEmpty()) {
+            return null;
+        }
+        return new TagContainsKeywordsPredicate(Arrays.asList(tagArgs.split("\\s+")));
+    }
+
+    private Predicate<Person> createPhonePredicate(String phoneArgs) {
+        if (phoneArgs.isEmpty()) {
+            return null;
+        }
+        return new PhoneNumberMatchesPredicate(phoneArgs);
```

Comment:
There seems to be code duplication here? Each function seems to differ only in Predicate created, can we combine them?